src/main/java/com/zendesk/maxwell/producer/MaxwellKafkaProducer.java

```diff
@@ -110,9 +111,21 @@ public MaxwellKafkaProducer(MaxwellContext context, Properties kafkaProperties,
 		this.inflightMessages = new InflightMessageList();
 	}
 
+	private Integer getNumPartitions(String topic) {
+		try {
+			return this.kafka.partitionsFor(topic).size(); //returns 1 for new topics
+		} catch(TimeoutException e) {
+			LOGGER.error("Topic '" + topic + "' name does not exist. Exception: " + e.getLocalizedMessage());
+			throw e;
```

Comment:
blows the shit out of the scope, but maybe there's an opportunity here; what about a smallish `kafka-metadata` jar?  Something where you could at first answer queries like: 

- what version of kafka am I on? 
- what topics exist?  

and then, maybe eventually trying to turn into the missing kafka-admin library (create/modify/delete topics programmatically, etc).

Lots of duplication/overlap with what the admin utils already do, but it might be worth it to see how easy they'd be to extract out into a JAR....

or am I a DIY nut?  unclear.